src/api/types.ts

```diff
@@ -1,6 +1,8 @@
 import { AxiosRequestConfig } from 'axios';
 
 export type Action = Required<Pick<AxiosRequestConfig, 'method' | 'url'>> &
+  Pick<AxiosRequestConfig, 'data' | 'headers'> &
```

Comment:
Czy to realnie coś dodaje? Bo w linii następnej i tak dodajesz wszystko z AxiosRequestConfig, poza trzema paramsami, więc i data i headers powinny też tam być dodane, co nie?